ethcore/evm/src/interpreter/mod.rs

```diff
@@ -843,6 +845,12 @@ impl<Cost: CostType> Interpreter<Cost> {
 			instructions::GASLIMIT => {
 				self.stack.push(ext.env_info().gas_limit.clone());
 			},
+			instructions::CHAINID => {
+				self.stack.push(ext.chain_id().into())
+			},
+			instructions::SELFBALANCE => {
+				self.stack.push(ext.balance(&self.params.origin)?);
```

Comment:
@sorpaas is `self.params.origin` what I need here, i.e. is that what corresponds to `this`?